src/neo4j-arc/cypher-language-support/cypher-editor/CypherEditor.tsx

```diff
@@ -385,11 +387,21 @@ export class CypherEditor extends React.Component<
       KeyMod.CtrlCmd | KeyCode.US_DOT,
       this.props.onDisplayHelpKeys
     )
-    this.editor.addCommand(
-      KeyCode.Escape,
-      this.props.toggleFullscreen,
-      '!suggestWidgetVisible && !findWidgetVisible'
-    )
+
+    function keys<T>(object: T) {
+      return Object.keys(object) as Array<keyof T>
+    }
+
+    keys(this.props.additionalCommands).forEach(key => {
+      const command = this.props.additionalCommands[key]
+      if (!command) {
+        return
+      }
+
+      this?.editor?.addCommand(key, () => {
+        command()
+      })
```

Comment:
```suggestion
      this.editor?.addCommand(key, command)
```

would avoid creating an extra function (if it works with the types)